x-pack/plugin/write-load-plugin/src/main/java/org/elasticsearch/xpack/writeload/WriteLoadForecastIndexSettingProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.writeload;
+
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.common.compress.CompressedXContent;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexSettingProvider;
+import org.elasticsearch.index.IndexSettings;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.function.BooleanSupplier;
+
+class WriteLoadForecastIndexSettingProvider implements IndexSettingProvider {
+    private final BooleanSupplier hasValidLicense;
+
+    WriteLoadForecastIndexSettingProvider(BooleanSupplier hasValidLicense) {
+        this.hasValidLicense = hasValidLicense;
+    }
+
+    @Override
+    public Settings getAdditionalIndexSettings(
+        String indexName,
+        String dataStreamName,
+        boolean timeSeries,
+        Metadata metadata,
+        Instant resolvedAt,
+        Settings allSettings,
+        List<CompressedXContent> combinedTemplateMappings
+    ) {
+        if (dataStreamName != null && metadata.dataStreams().get(dataStreamName) != null && hasValidLicense.getAsBoolean()) {
```

Comment:
This is where we decider if we can add the _forecasted_ write load depending on the current license. It's a bit convoluted but I couldn't find a better way to hook into the rollover.

Reply:
One option is to add a new method to `ClusterPlugin` that returns a `WriteLoadHandler`, which can then be implemented in the plugin? This can then have the 2-3 methods necessary (calculate write-load, get write-load).